compatibility with jsonform

fuhrysteve opened the pull request:
Thanks for this great, handy little library. I can't imagine you'd want to accept this pull request as-is, since it breaks backwards compatibility pretty severely..

If you have any suggestions, however, for how to accomplish what I'm trying to do here in a more flexible way (I'm essentially rendering a more complete json-schema output), then let me know (presuming this project is still active).

In any case, this seems to accomplish what I'm looking to do and will work just fine for me.

Thanks!

fuhrysteve committed:
gitignore

fuhrysteve committed:
jsonform compatible output

See: https://github.com/joshfire/jsonform

fuhrysteve committed:
don't alter class attributes!

fuhrysteve committed:
use isinstance instead of __name__, and use deepcopy

fuhrysteve committed:
Password support

fuhrysteve committed:
fix TypeError & support json-form

zbyte64 commented:
Your explicit matching of field classes is the proper way forward imho. If there are more projects like `wtforms_components` that would make sense to integrate with then we would want a module folder with an import for each integration. As for flexibility, conversions could allow for callables to be registered as well as dictionaries. When it comes across a callable, it will be called with the field being converted and the return will be treated as attributes to merge.

JasonMRojas committed:
Add length to front end

fuhrysteve committed:
Merge pull request #1 from JasonMRojas/lengthFront

Add length to front end

fuhrysteve committed:
Update wtforms-jsonschema for WTForms 3.x compatibility

- Remove html5 field imports (moved to wtforms.fields in WTForms 3.x)
- Remove six dependency (Python 2 compat no longer needed)
- Bump wtforms requirement to >=3 in setup.py

fuhrysteve committed:
Merge pull request #3 from MontmereLimited/wtfforms3

Update wtforms-jsonschema for WTForms 3.x compatibility
